### PR TITLE
Add Contact Steps, Implemented via Mixin

### PR DESCRIPTION
**Pull request by amcpanaligan**
1. Implemented Contact SObject through Typescript `mixins`
2. The `grpc-server` needs to be modified to apply mixin. See `line 24`
Edit:
3. Tests for `steps` are up. CC now at 100%

Please review

### CODE REVIEW

File: src/core/grpc-server.ts [hunk 1]
```diff
@@ -20,7 +21,9 @@ if (process.env.USE_SSL) {
   credentials = grpc.ServerCredentials.createInsecure();
 }
 
-server.addService(CogService, new Cog(ClientWrapper));
+const client = ContactMixin(ClientWrapper);
```
Comment (iamEAP) [inline src/core/grpc-server.ts]: Hmm...  I wonder if it's possible to perform the actual mixin "mixing" within the `client-wrapper.ts` itself.  E.g. the final exported class contains all of the methods, but individual groups of methods would come from mixin classes.  

For example...

```typescript
// src/client/client-wrapper.ts pseudo-code

import { ContactAwareMixin } from './mixins/contact-aware.ts';
import { LeadAwareMixin } from './mixins/lead-aware.ts';
class ClientWrapper {
  constructor(auth: grpc.Metadata, clientConstructor = jsforce) {/*...*/}
  /*...*/
}

interface ClientWrapper extends ContactAwareMixin, LeadAwareMixin  {};

applyMixins(ClientWrapper, [ContactAwareMixin, LeadAwareMixin]);

export ClientWrapper as ClientWrapper
```

```typescript
// src/client/mixins/contact-aware.ts pseudo-code

export class ContactAwareMixin {
  public async createContact(contact) {/*...*/}
  /*...*/
}
```

See: https://www.typescriptlang.org/docs/handbook/mixins.html

Comment (amcpanaligan) [reply src/core/grpc-server.ts]: They say that the example from the typescript _official_ are not up to date. Though the `applyMixins` function is looking clean. Even if it's outdated, should we go with it?

Comment (amcpanaligan) [reply src/core/grpc-server.ts]: By go with it. Completely follow their guide.

Comment (iamEAP) [reply src/core/grpc-server.ts]: Let's give it a shot and see what happens...? 😬

**Comment by amcpanaligan** [comment]
Working..

File: src/core/grpc-server.ts (hunk 1 above)
Comment (amcpanaligan) [reply src/core/grpc-server.ts]: New change pushed. Let me know what you think. If it's good then I will teach @saludangelito on how to convert his work to `mixins`. Thanks for all your help!

**Comment by iamEAP** [comment]
This is looking great.  Thanks @amcpanaligan!

**Pull request merged**